hadoop-hdds/interface-admin/src/main/proto/ScmAdminProtocol.proto

```diff
@@ -209,6 +221,72 @@ message ContainerRequestProto {
   optional string traceID = 5;
 }
 
+message ECContainerRequestProto {
+  required ECReplicationConfig ecReplicationConfig = 1;
+  required string owner = 2;
+  optional string traceID = 3;
+}
+
+message ECContainerResponseProto {
+  enum Error {
+    success = 1;
+    errorContainerAlreadyExists = 2;
+    errorContainerMissing = 3;
+    scmNotLeader = 4;
+  }
+  required Error errorCode = 1;
+  required ContainerWithPipeline containerWithPipeline = 2;
+  optional string errorMessage = 3;
+}
+
+message GetEcContainerRequestProto {
+  required int64 containerID = 1;
+  optional string traceID = 2;
+
+}
+
+message GetEcContainerResponseProto {
+  required ContainerInfoProto containerInfo = 1;
+}
+
+message GetEcContainerWithPipelineRequestProto {
+  required int64 containerID = 1;
+  optional string traceID = 2;
+
+}
+
+message GetEcContainerWithPipelineResponseProto {
+  required ContainerWithPipeline containerWithPipeline = 1;
+}
+
+message GetEcContainerReplicasRequestProto {
+  required int64 containerID = 1;
+  optional string traceID = 2;
+}
+
+message GetEcContainerReplicasResponseProto {
+  repeated SCMContainerReplicaProto containerReplica = 1;
+}
+
+message GetEcContainerWithPipelineBatchRequestProto {
+  repeated int64 containerIDs = 1;
+  optional string traceID = 2;
+}
+
+message GetExistEcContainerWithPipelinesInBatchResponseProto {
+  repeated int64 containerIDs = 1;
+  optional string traceID = 2;
+}
+
+message GetExistEcContainerWithPipelinesInBatchRequestProto {
+  repeated int64 containerIDs = 1;
+  optional string traceID = 2;
+}
+
+message GetEcContainerWithPipelineBatchResponseProto {
+  repeated ContainerWithPipeline containerWithPipelines = 1;
+}
+
```

Comment:
I don't think these new requests and responses are needed.  Add `optional ECReplicationConfig ecReplicationConfig` to existing `ContainerRequestProto`, and use `hasEcReplicationConfig()` to decide if it's EC or not.

Reply:
We can change factor to `optional`, but should still set some value (in case the receiver is old version, assuming `required`).  It can be any valid value, as long as it is ignored for EC.